driver/src/main/java/org/neo4j/driver/internal/SecuritySettings.java

```diff
@@ -59,7 +60,20 @@ public boolean encrypted()
 
     private boolean isCustomized()
     {
-        return this != DEFAULT;
+        return !(DEFAULT.encrypted() == this.encrypted() && DEFAULT.hasEqualTrustStrategy( this ));
```

Comment:
Indeed this is the source of contention across serialization exercises. This change would fix that.